### PR TITLE
Add Uncentered version of RMSProp

### PR DESCRIPTION
**Pull request by leezu**
See the discussion at https://github.com/dmlc/mxnet/pull/5043#issuecomment-280674628

This supersedes #5075 as I erroneously created that pull request based on my master branch. This pull request already contains the refactored python API for RMSProp as discussed in #5075.

### CODE REVIEW

File: tests/python/unittest/test_optimizer.py [hunk 1]
```diff
@@ -196,14 +211,24 @@ def update(self, index, weight, grad, state):
         lr = self._get_lr(index)
         wd = self._get_wd(index)
         self._update_count(index)
-        n, g, delta = state
         grad = grad * self.rescale_grad
```
Comment (sxjscience) [inline tests/python/unittest/test_optimizer.py]: As described in https://github.com/dmlc/mxnet/issues/5099, the old way of handling weight decay is not correct. We may need to use `grad = grad * self.rescale_grad + wd * weight`.

File: tests/python/unittest/test_optimizer.py [hunk 2]
```diff
@@ -216,7 +241,14 @@ def test_rms():
               {'rescale_grad': 0.8},
               {'clip_gradient': 0.5, 'wd': 0.07},
               {'clip_gradient': 0.4, 'rescale_grad': 0.14, 'wd': 0.03},
-              {'rescale_grad': 0.8, 'wd': 0.05}]
+              {'rescale_grad': 0.8, 'wd': 0.05},
+              {'centered': True},
```
Comment (piiswrong) [inline tests/python/unittest/test_optimizer.py]: add test for centered =False?

Comment (leezu) [inline tests/python/unittest/test_optimizer.py]: centered=False is the default, so it is already addressed.

**Comment by leezu** [comment]
The changes proposed in this pull request are also part of #5137 and will be discussed there.

**Pull request closed without merging**